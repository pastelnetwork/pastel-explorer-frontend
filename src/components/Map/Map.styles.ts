import styled from 'styled-components';

import { Card as MuiCard, CardContent as MuiCardContent } from '@mui/material';

export const MapContainer = styled.div`
  height: 340px;

  @media screen and (min-width: 1024px) and (max-width: 1280px) {
    height: 258px;
  }
`;

export const Card = styled(MuiCard)`
  background: transparent;
  box-shadow: none;

  h4 {
    margin: 0;
    padding-top: 18px;
    padding-bottom: 18px;
    padding-left: 16px;
    background: ${props => props.theme.card.titleColor};
  }
`;

export const CardContent = styled(MuiCardContent)`
  &:last-child {
    padding-top: 0;
    padding-bottom: ${props => props.theme.spacing(4)}px;
  }
`;
