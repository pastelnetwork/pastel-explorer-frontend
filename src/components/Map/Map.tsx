import React from 'react';
import { VectorMap } from 'react-jvectormap';

import * as Styles from './Map.styles';
import { defaultMapOptions } from './Map.options';
import './map.css';

interface MarkerStyle {
  fill: string;
  stroke: string;
}

export interface MarkerData {
  type: string;
  country: string;
  city: string;
  id: string;
  ip: string;
}

export interface MarkerProps {
  latLng: [number, number];
  name: string;
  style: MarkerStyle;
  data: Array<MarkerData>;
}

export interface MapProps {
  title: string | React.ReactNode;
  markers: Array<MarkerProps> | null;
  options?: typeof VectorMap;
}

const Map: React.FC<MapProps> = ({ title, markers, options }) => {
  return (
    <Styles.Card>
      <h4>{title}</h4>
      <Styles.CardContent>
        <Styles.MapContainer>
          {markers && <VectorMap markers={markers} {...defaultMapOptions} {...options} />}
        </Styles.MapContainer>
      </Styles.CardContent>
    </Styles.Card>
  );
};

Map.defaultProps = {
  options: undefined,
};

export default Map;
