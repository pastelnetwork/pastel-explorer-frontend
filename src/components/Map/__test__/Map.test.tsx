import { shallow } from 'enzyme';

import 'jest-styled-components';

import Map, { MarkerProps } from '../Map';

describe('components/Map', () => {
  const markers = [
    {
      latLng: [39.9625, -83.0061],
      name: 'United States - Columbus (Peer(s): 1, Supernode(s): 1)',
      style: {
        fill: '#000',
        stroke: '#279989',
      },
      data: [
        {
          type: 'Peer',
          country: 'United States',
          city: 'Columbus',
          id: 'f34a34b3-bf49-4323-8cdf-34c270dd1bec',
          ip: '18.118.218.206',
        },
        {
          type: 'Supernode',
          country: 'United States',
          city: 'Columbus',
          id: '8c939e2b-09c7-4df1-a034-c951bfe8c695',
          ip: '52.15.244.16',
        },
      ],
    },
    {
      latLng: [51.1878, 6.8607],
      name: 'Germany - Düsseldorf (Peer(s): 2)',
      style: {
        fill: '#279989',
        stroke: '#279989',
      },
      data: [
        {
          type: 'Peer',
          country: 'Germany',
          city: 'Düsseldorf',
          id: 'c4b9a89e-1736-4c3e-ae11-9813a055e97a',
          ip: '38.242.134.66',
        },
        {
          type: 'Peer',
          country: 'Germany',
          city: 'Düsseldorf',
          id: 'f2d4b69c-c9c9-48b3-8e62-0f65cba444b2',
          ip: '109.205.181.92',
        },
      ],
    },
  ];
  const wrapper = shallow(
    <Map title="Lorem ipsum dolor sit amet" markers={markers as MarkerProps[]} />,
  );

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
