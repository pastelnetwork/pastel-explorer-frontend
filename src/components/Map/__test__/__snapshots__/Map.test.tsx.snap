// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Map renders correctly 1`] = `ShallowWrapper {}`;
